src/main/java/org/folio/service/RequestExpirationService.java

```diff
@@ -47,51 +47,51 @@
 import io.vertx.sqlclient.Row;
 import io.vertx.sqlclient.RowSet;
 
-public class ExpirationTool {
+public class RequestExpirationService {
   private static final Logger log = LogManager.getLogger();
   private static final String JSONB_COLUMN = "jsonb";
-
-  private ExpirationTool() {
-    //do nothing
+  private final String idFieldName;
+  private final Function<Request, String> idExtractor;
+  private final PostgresClient pgClient;
+  private final EventPublisherService eventPublisherService;
+
+  public RequestExpirationService(Map<String, String> okapiHeaders, Vertx vertx,
+    String idFieldName, Function<Request, String> idExtractor) {
+
+    this.idFieldName = idFieldName;
+    this.idExtractor = idExtractor;
+    pgClient = PostgresClient.getInstance(vertx, okapiHeaders.get(TENANT_HEADER));
+    eventPublisherService = new EventPublisherService(vertx, okapiHeaders);
   }
 
-  public static Future<Void> doRequestExpirationForTenant(Map<String, String> okapiHeaders, Vertx vertx) {
+  public Future<Void> doRequestExpiration() {
     Promise<Void> promise = Promise.promise();
-
-    String tenant = okapiHeaders.get(TENANT_HEADER);
-
-    PostgresClient pgClient = PostgresClient.getInstance(vertx, tenant);
-
     List<JsonObject> context = new ArrayList<>();
 
-    pgClient.startTx(conn -> getExpiredRequests(conn, vertx, tenant)
-      .compose(requests -> closeRequests(conn, vertx, tenant, requests, context))
-      .compose(itemIds -> getOpenRequestsByItemIds(conn, vertx, tenant, itemIds))
-      .compose(requests -> reorderRequests(conn, vertx, tenant, requests))
+    pgClient.startTx(conn -> getExpiredRequests(conn)
+      .compose(expiredRequests -> closeRequests(conn, expiredRequests, context, idExtractor)
```

Comment:
no need to pass `idExtractor` as parameter, it is accessible as an instance variable. Also, can we think of a better name for `context` variable?